### PR TITLE
pkg/util/iptables missing unit tests

### PR DESCRIPTION
**Pull request by nebril**
Added tests for Save, SaveAll, Restore and RestoreAll

### CODE REVIEW

File: pkg/util/iptables/iptables_test.go [hunk 1]
```diff
@@ -771,3 +771,224 @@ func TestReload(t *testing.T) {
...
+			func(cmd string, args ...string) exec.Cmd { return exec.InitFakeCmd(&fcmd, cmd, args...) },
+			func(cmd string, args ...string) exec.Cmd { return exec.InitFakeCmd(&fcmd, cmd, args...) },
+		},
+	}
+	runner := New(&fexec, dbus.NewFake(nil, nil), ProtocolIpv4)
+	defer runner.Destroy()
+	// Success.
+	o, err := runner.Save(TableNAT)
+	if err != nil {
+		t.Errorf("expected success, got %v", err)
```
Comment (xiangpengzhao) [inline pkg/util/iptables/iptables_test.go]: `t.Errorf` should be `t.Fatalf` because `o` may be nil. `TestSaveAll` has the same issue.

Comment (nebril) [inline pkg/util/iptables/iptables_test.go]: Makes sense, will do

**Comment by nebril** [comment]
@k8s-bot gci gke e2e test this

**Comment by nebril** [comment]
@k8s-bot cvm gke e2e test this

**Comment by nebril** [comment]
@k8s-bot cvm gke e2e test this

**Comment by nebril** [comment]
@k8s-bot gci gke e2e test this

File: pkg/util/iptables/iptables_test.go [hunk 2]
```diff
@@ -771,3 +771,224 @@ func TestReload(t *testing.T) {
...
+	}
+	runner := New(&fexec, dbus.NewFake(nil, nil), ProtocolIpv4)
+	defer runner.Destroy()
+	// Success.
+	o, err := runner.Save(TableNAT)
+	if err != nil {
+		t.Fatalf("expected success, got %v", err)
+	}
+
+	if string(o[:len(output)]) != output {
```
Comment (thockin) [inline pkg/util/iptables/iptables_test.go]: why the `[:len...]` rather than just `string(o)` ?

**Review by thockin — APPROVED** [review]
thanks!

**Comment by smarterclayton** [comment]
/approve

on tim's recognizance

**Comment by thockin** [comment]
@k8s-bot cvm gke e2e test this

@k8s-bot kubemark e2e test this

**Pull request merged**